packages/api/src/utils/db-client.js

```diff
@@ -554,19 +563,66 @@ export class DBClient {
         'uploads_multipart_total',
       ])
 
-    if (error) {
-      throw new DBError(error)
-    }
+    const dagByteSizeQuery = metricsQuery
+      .select('name, dimensions, value, collected_at')
+      .match({
+        name: 'dagcargo_project_bytes_in_active_deals',
+        dimensions: '{{project,nft.storage}}',
+      })
+      .order('collected_at', { ascending: true })
+      .single()
 
-    if (!data || !data.length) {
-      return undefined
-    }
+    const d = new Date()
+    d.setDate(d.getDate() - 7)
 
-    return data.reduce((obj, curr) => {
+    const dagByteSizeHistory = metricsLogQuery
+      .select('name, dimensions, value, collected_at')
+      .match({
+        name: 'dagcargo_project_bytes_in_active_deals',
+        dimensions: '{{project,nft.storage}}',
+      })
+      .lte('collected_at', d.toISOString())
+      .order('collected_at', { ascending: false })
+      .single()
+
+    try {
+      const fetchAllMetrics = await Promise.all([
```

Comment:
Personally I'd destructure here and access `.data` or `.error` when needed e.g.

```suggestion
      const [primaryRes, dagSizeRes, dagSizeHistRes] = await Promise.all([
```

Reply:
Fair enough! fixed